src/parser/expression.js

```diff
@@ -1014,6 +1022,15 @@ pp.parseIdentifier = function (liberal) {
   return this.finishNode(node, "Identifier");
 };
 
+pp.parsePrivateName = function (liberal) {
+  const isPrivate = this.eat(tt.hash);
+  const node = this.parseIdentifier(liberal);
```

Comment:
I'm not talking about naming convention, I'm talking about grammar. The grammar for Identifier is more restrictive than IdentifierName/property names, in that it doesn't allow keywords. The grammar for private state specifically allows keywords as the name (just like ordinary properties).